tests/integration/rest/diddoc/diddoc/negative_test.go

```diff
@@ -0,0 +1,235 @@
+//go:build integration
+
+package diddoc
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	testconstants "github.com/cheqd/did-resolver/tests/constants"
+	utils "github.com/cheqd/did-resolver/tests/integration/rest"
+
+	"github.com/cheqd/did-resolver/types"
+	"github.com/go-resty/resty/v2"
+	. "github.com/onsi/ginkgo/v2"
+
+	. "github.com/onsi/gomega"
+)
+
+var _ = DescribeTable("Negative: Get DIDDoc", func(testCase utils.NegativeTestCase) {
+	client := resty.New()
+
+	resp, err := client.R().
+		SetHeader("Accept", testCase.ResolutionType).
+		Get(testCase.DidURL)
+	Expect(err).To(BeNil())
+	Expect(testCase.ExpectedStatusCode).To(Equal(resp.StatusCode()))
+
+	var receivedDidResolution types.DidResolution
+	Expect(json.Unmarshal(resp.Body(), &receivedDidResolution)).To(BeNil())
+
+	expectedDidResolution := testCase.ExpectedResult.(types.DidResolution)
+	utils.AssertDidResolution(expectedDidResolution, receivedDidResolution)
+},
+
+	Entry(
+		"cannot get DIDDoc with an existent DID, but not supported ResolutionType",
+		utils.NegativeTestCase{
+			DidURL: fmt.Sprintf(
+				"http://localhost:8080/1.0/identifiers/%s",
+				testconstants.UUIDStyleMainnetDid,
+			),
+			ResolutionType: string(types.JSON),
+			ExpectedResult: types.DidResolution{
+				Context: "",
+				ResolutionMetadata: types.ResolutionMetadata{
+					ContentType:     types.JSON,
+					ResolutionError: "representationNotSupported",
+					DidProperties:   types.DidProperties{},
+				},
+				Did:      nil,
+				Metadata: types.ResolutionDidDocMetadata{},
+			},
+			ExpectedStatusCode: http.StatusNotAcceptable,
```

Comment:
Yes, we are sending "Accept-Encoding" by default - `gzip, deflate, br`. When we send that Accept-Encoding (`gzip, deflate, br`) then DID Resolver returns Encoding-Type - `gzip` because we only support it for now.